### DISCUSSION TITLE
External access using Zoraxy as reverse proxy

### DISCUSSION THREAD

**Discussion by freedbygrace** [docmost/docmost#287] [Help / Questions]
Everything works fantastic internally, but when accessing over my reverse proxy which has many other services exposed the same way, this application just shows a blank white screen. Can any assistance or guidance be provided because out of all the wikis and note taking apps, this is absolutely the best IMHO, so its a bummer I could not get it to work externally straight away.

I will upload my Docker Compose and Sample Env File for contribution (All sensitive information has been replaced with examples for a zero touch experience.)

![image](https://github.com/user-attachments/assets/d19da90f-3806-492a-ac70-40a9d03bc3e6)

![image](https://github.com/user-attachments/assets/fa7874c8-c97c-479b-a688-66352ade3c9c)

![image](https://github.com/user-attachments/assets/744c1405-c109-43ca-90c0-68ff2bb5b2c6)

[docker-compose.txt](https://github.com/user-attachments/files/16919741/docker-compose.txt)
[docker-compose-env.txt](https://github.com/user-attachments/files/16919742/docker-compose-env.txt)

[_DOCMOST-APP-001_logs.txt](https://github.com/user-attachments/files/16919748/_DOCMOST-APP-001_logs.txt)

**Comment by Philipinho** [comment]
Please send screenshots of your browser's network and console tab.

**Comment by AX-AMote** [comment]
Hey there! Thanks for the reply! Screenshots are coming your way!

![image](https://github.com/user-attachments/assets/e5537cd0-91fa-463a-aa60-5b9c71707ad7)
![image](https://github.com/user-attachments/assets/a9517e35-cf71-414b-891b-654a25df3db8)

**Comment by GamingForLive** [comment]
Nice seeing someone else use Zoraxy but im experiencing the Exact same Issues too.
 
 FYI: My proxy is configured to Skip the WebSocket Origin Check (Default setting in Zoraxy) And ive added no aditional headders
 (no added Screenshot as my browser console looks 1:1 exactly like the one from AX-Mote)

**Reply by ExXxtr3me** [reply]
I'm also using Zoraxy and have the same problem, but it worked about a month ago. I also tried it with NPM instead of Zoraxy but it has the same problem.

**Comment by Morethanevil** [comment]
Okay this is why I got error 415 on my VPS. I use Zoraxy too. Same error if I use IPV6 on my local machine for docmost. IPV4 works fine locally

Tried a lot. My issue was [this one](https://github.com/docmost/docmost/issues/67)

Maybe related

**Comment by xmeadow** [comment]
Funny, I tried to switch from NPM to Zoraxy. First Proxy in my setup was docmost. A google search later "zoraxy white page" brought me directly back to this discussion in the docmost-repo although I didn't think it was docmost's fault.
Well, I switched back to NPM and Docmosts works like a charm.

**Reply by xmeadow** [reply]
Here my screenshots if it helps. I hope one day sombody finds a solution.
![grafik](https://github.com/user-attachments/assets/f153942f-a4db-4b62-9fa5-0c2c06272144)
![grafik](https://github.com/user-attachments/assets/17579767-06d1-497d-84b6-3bcd3e3fb293)

**Comment by GamingForLive** [comment]
yea idk im currently on another proxy thats closer to a waf so the issue is basically resolved

**Reply by xmeadow** [reply]
I have not found another zoraxy-user with a resolved issue. ;) But I think it may be a zoraxy issue and not a docmost issue - something with websockets I don't understand so I can't file a issue at zoraxy.

**Comment by johnny15243** [comment]
Hi there.

I had the same issue trying netbird.

The problem is.
Docmost makes an h2 connection. You can see it in @xmeadow screenshot.

There are three different connection types:
| Type | description |
|---------|--------------------|
| h1 | http1 |
| h2 | http2 with TLS |
| h2c | http2 without TLS |

I do ignore the upcoming QUIC protokoll for the moment.

Zeroxy implemented h1 and h2.
In my opinion http2 should always have an TLS encryption. So I think docmost needs to change and give you the possibility to run there service behind your proxy with an TLS encrypted http/2 connection. (h2)

But I think it would be nice if zoraxy adds the h2c connection.

See my initial netbird/zoraxy problem and the request for adding h2c to zoraxy [GitHub](https://github.com/tobychui/zoraxy/issues/381)

So back to docmost.
Docmost runs http. So you have to setup an h2c connection in your reverse proxy to bring up.
If docmost won't give you the possibility to run there service as TLS encrypted H2 connection, you have to choose a different reverseproxy.
Caddy should support http/2 without TLS encryption (h2c).

I did not try docmost in combination with zoraxy, so its only what I found from there [configurations](https://docmost.com/docs/self-hosting/environment-variables)

So please correct me if I'm wrong.


Here a short explanation from chatgpt about the difference between H2 and h2c

```
The difference between H2 and h2c lies in how they handle the connection:

H2 refers to HTTP/2 over TLS (Transport Layer Security), meaning it uses a secure, encrypted connection (similar to HTTPS).
h2c refers to HTTP/2 over cleartext, meaning it uses a plain, unencrypted connection (similar to HTTP).
In summary:

H2 = secure version (encrypted)
h2c = non-secure version (unencrypted)
```

**Reply by Morethanevil** [reply]
I wrote an email to the dev of Zoraxy. He will look into this next month. I linked your comment 😊

**Reply by Morethanevil** [reply]
A [PR for h2c](https://github.com/tobychui/zoraxy/commit/07dc63a82cd50b2e378ea390eff878735f40a615) was made ☝🏻

**Reply by DerLeole** [reply]
Do I understand this correctly that Docmost can answer in both http1 and http2 (because nestjs/fastify support this by default) depending on the request, but only h2c because there are no certs used within.

If the recommended procedure is to use a reverse proxy anyways to serve docmost via https, would the conclusion to that not be to simply force Docmost to only respond in http1 regardless of request, like expected without any ssl certs?

**Comment by clodobox** [comment]
Hello there,

I switched from Zoraxy to Nginx Proxy Manager and managed to connect !

Unfortunately, I think that Zoraxy still doesn't have all the options to work around the problem.

What got me thinking was that there's [a tutorial](https://mariushosting.com/how-to-install-docmost-on-your-synology-nas/) on the “Marius Hosting” blog for installing Docmost on a Synology with built-in reverse proxy.
The built-in Synology reverse proxy interface only edits the Nginx config file, so there's no reason why it shouldn't work with NPM.

Here's the configuration I used in NPM:
![image](https://github.com/user-attachments/assets/f9399f35-5b81-4e95-a540-92a14a76dc9b)
![image](https://github.com/user-attachments/assets/39119d5e-f7c5-467c-88b1-3a40861d858a)

It may be that the configuration is not optimal, I'm not an expert in this field.
Please correct me if you see any weirdness.

**Comment by DerLeole** [comment]
@Morethanevil Looks like the PR was successfully merged in the 3.1.7 release of Zoraxy today. However either I don't know how to activate the new experimental h2c support, or the feature is still broken, or it was another problem that causes Docmost to not work with Zoraxy, as the error persists.

Any ideas?

**Reply by Morethanevil** [reply]
I write an email to Toby (the dev), he should it :)

I was looking for this option too, but did not find it

**Reply by Morethanevil** [reply]
I got a reply this morning and just read it. In the meantime you opened an issue, let us continue there. It is a problem with Zoraxy :)

For anyone interested, [here is the issues](https://github.com/tobychui/zoraxy/issues/527)

**Comment by tobychui** [comment]
Hi there,

I am having the same issue here. On the left is the one proxy through Zoraxy and the right is the request when I directly request to Docmost.
![圖片](https://github.com/user-attachments/assets/15e627fa-9d61-4c1d-aacd-be89d49c44d8)

Not sure why Docmost didn't send some of the headers when proxying through Zoraxy. Other static resources like `/favicon-16x16.png` do works without an issue. I am guessing it is a front-end related issue?

**Reply by mrcsdl** [reply]
got the same issue. Localy it works fine as soon as i turn on zoraxy and proxy the traffic i just get a white page :D

**Reply by tobychui** [reply]
@mrcsdl Thanks for the feedback, just in case you didn't notice, I am the author of Zoraxy and I am writing just hope that the author or maintainer of docmost provide me some hints on this. Technically speaking I am not sure if h2c is actually the issue here as static resources works but not the UI is weird.

**Reply by mrcsdl** [reply]
yeah i know. ill love zoraxy and also opend some issues there :D

**Comment by avdept** [comment]
Any updates on this?

**Reply by Morethanevil** [reply]
The dev of Zoraxy, needs help of the docmost dev, as he said [in this thread](https://github.com/docmost/docmost/discussions/287#discussioncomment-12250136) already. 

For now, you can use [my workaround](https://github.com/tobychui/zoraxy/wiki/Troubleshooting----Workarounds#5-docmost-workaround-for-zoraxy) from the Zoraxy wiki.

**Reply by sbeneke** [reply]
After a few months of experimenting, I checked "Disable Chunked Transfer Encoding" in Zoraxy v.3.2.5 today and now Docmost works for me via Zoraxy.
<img width="810" height="448" alt="image" src="https://github.com/user-attachments/assets/78c342c9-43d2-4a5c-9d0d-14ba718c52dd" />

**Reply by avdept** [reply]
wow nice find, thanks for sharing mate, appreciate your help. Time to get rid of nginx hack

**Reply by Daimonion1980** [reply]
Works also in my case. Thanks for this finding!
